sdk/src/instruction.rs

```diff
@@ -3,96 +3,124 @@
 use crate::{pubkey::Pubkey, short_vec, system_instruction::SystemError};
 use bincode::serialize;
 use serde::Serialize;
+use thiserror::Error;
 
 /// Reasons the runtime might have rejected an instruction.
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[derive(Serialize, Deserialize, Debug, Error, PartialEq, Eq, Clone)]
 pub enum InstructionError {
     /// Deprecated! Use CustomError instead!
     /// The program instruction returned an error
+    #[error("generic instruction error")]
     GenericError,
 
     /// The arguments provided to a program instruction where invalid
+    #[error("invalid instruction argument")]
     InvalidArgument,
 
     /// An instruction's data contents was invalid
+    #[error("invalid instruction data")]
     InvalidInstructionData,
 
     /// An account's data contents was invalid
+    #[error("invalid account data for instruction")]
```

Comment:
Do we need "for instruction" on the end of these?

Reply:
Since these are for implementing `Display` I wanted to hint back to `InstructionError` for the developers sake.  Happy to kill them if you don't think it'll be a problem

Reply:
Ok, yeah I figured that, leave it